src/IROperator.h

```diff
@@ -1404,6 +1404,24 @@ inline Expr memoize_tag(Expr result, Expr a, Expr b, Expr c, Expr d, Expr e, Exp
 }
 // @}
 
+/** Expressions tagged with this intrinsic are considered to be part
+ * of the steady state of some loop with a nasty beginning and end
+ * (e.g. a boundary condition). When Halide encounters likely
+ * intrinsics, it splits the containing loop body into three, and
+ * tries to simplify down all conditions that lead to the likely. For
+ * example, given the expression: select(x < 1, bar, x > 10, bar,
+ * likely(foo)), Halide will split the loop over x into portions where
+ * x < 1, 1 <= x <= 10, and x > 10.
+ *
+ * You're unlikely to want to call this directly. You probably want to
```

Comment:
This is an interesting point. If someone tags the wrong branch as likely things could go horribly wrong, but I guess that is like saying if someone chooses a really bad schedule things will be really slow. But should we hide the use of likely from the outside world more explicitly?